bindings/pyroot_experimental/PyROOT/python/ROOT/pythonization/_tfile.py

```diff
@@ -15,12 +15,71 @@
 
 from libROOTPython import AddFileOpenPyz
 from ROOT import pythonization
+import sys, errno, os
+from functools import partial
+
+# We check the Python version because FileNotFoundError
+# is not available in Python2
+def RaiseError(file_name):
+    from cppyy.gbl import gSystem
+    if isinstance(file_name, str) and gSystem.AccessPathName(file_name):
+        if sys.version_info > (3,0):
+            raise FileNotFoundError(errno.ENOENT,
+                    os.strerror(errno.ENOENT),
+                    file_name)
+        else:
+            raise IOError(errno.ENOENT,
+                    os.strerror(errno.ENOENT),
+                    file_name)
+
+def _TFileConstructor(self, *args):
+    # Redefinition of ROOT.TFile(str, ...):
+    # checks if the a file with the given name exists and raises
+    # a FileNotFoundError-like if not.
+    # After the check, the method is executed anyway and the
+    # value returned even if the file does not exist; this is done
+    # in order to be consistent with the C++ behavior.
+    # Parameters:
+    # self: instance of TFile class
+    # *args: arguments passed to the constructor; only args[0]
+    # (string containing the name of the file) is mandatory
+
+    # Check the existance of the file (and fail) only in READ
+    # mode
+    if len(args) == 1 or (len(args) == 2 and args[1] == 'READ'):
+        RaiseError(args[0])
+    tfile_instance = self._OriginalConstructor(*args)
+    return tfile_instance
+
+def _TFileOpen(klass, *args):
+    # Redefinition of ROOT.TFile.Open(str, ...):
+    # checks if the a file with the given name exists and raises
+    # a FileNotFoundError-like if not.
+    # After the check, the method is executed anyway and the
+    # value returned even if the file does not exist; this is done
+    # in order to be consistent with the C++ behavior.
+    # Parameters:
+    # klass: TFile class
+    # *args: arguments passed to the constructor; only args[0]
+    # (string containing the name of the file) is mandatory
+
+    # Check the existance of the file (and fail) only in READ
+    # mode
+    if len(args) == 1 or (len(args) == 2 and args[1] == 'READ'):
+        RaiseError(args[0])
+    tfile_instance = klass._OriginalOpen(*args)
```

Comment:
Should we also check the result? (i.e, that it is neither ```nullptr``` nor ```tfile_instance->IsZombie();```

Reply:
I thought to only add the "raising-error" behavior on top without checking or touching the value returned because maybe there are tests from the experiments that rely on objects having `IsZombie() = True` or being `nullptr`, but fur sure I can change it.

Reply:
humm .. but the ' add the "raising-error" behavior on top ...' is already changing the behavior ... just in a subset of the error cases ...